external/sql/storm-sql-core/src/test/org/apache/storm/sql/compiler/backends/standalone/TestPlanCompiler.java

```diff
@@ -71,9 +71,9 @@ public void testLogicalExpr() throws Exception {
 
   @Test
   public void testNested() throws Exception {
-    String sql = "SELECT ID, MAPFIELD, NESTEDMAPFIELD, ARRAYFIELD " +
+    String sql = "SELECT ID, MAPFIELD['c'], NESTEDMAPFIELD, ARRAYFIELD " +
             "FROM FOO " +
-            "WHERE NESTEDMAPFIELD['a']['b'] = 2 AND ARRAYFIELD[1] = 200";
```

Comment:
Like I said before (PR desc and comment), with Calcite Janino compiler it is not possible. (even with explicit type information)
Let's wait for the response from Calcite dev@ list.